File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int n) {
    string answer = "";
    int rem = 0;
    while(n != 0) {
        rem = n % 3, n /= 3;
        if (rem == 0) {
            answer = "4" + answer;
            n--;
        } else {
            answer = to_string(rem) + answer;
        }
    }
    return answer;
}

File: 프로그래머스/2/12899. 124 나라의 숫자/124 나라의 숫자.java
class Solution {
    
    public String solution(int n) {
        String[] d = {"4", "1", "2"};
        StringBuilder sb = new StringBuilder();
        while (n > 0) {
            sb.insert(0, d[n % 3]);
            n = (n - 1) / 3;
        }
        return sb.toString();
    }
}